aws/resource_aws_workspaces_directory.go

```diff
@@ -225,6 +238,21 @@ func resourceAwsWorkspacesDirectoryUpdate(d *schema.ResourceData, meta interface
 		log.Printf("[DEBUG] WorkSpaces Directory %q self-service permissions are set", d.Id())
 	}
 
+	if d.HasChange("ip_group_ids") {
+		ipGroupIds := d.Get("ip_group_ids").(*schema.Set).List()
```

Comment:
same as above, schema struct has diff function is remove of old groups is needed by calling d.GetChange

Reply:
```
		o, n := d.GetChange("endpoint_configuration.0.vpc_endpoint_ids")
		prefix := "/endpointConfiguration/vpcEndpointIds"

		old := o.(*schema.Set).List()
		new := n.(*schema.Set).List()
```

Reply:
I have been looking for something like this, I'll add IP group disassociation with it.